Src/GoogleApis.Auth/OAuth2/Flows/GoogleAuthorizationCodeFlow.cs

```diff
@@ -39,6 +40,10 @@ public class GoogleAuthorizationCodeFlow : AuthorizationCodeFlow
 
         /// <summary>Gets or sets the include granted scopes indicator.</summary>
         public bool? IncludeGrantedScopes { get { return includeGrantedScopes; } }
+        private readonly KeyValuePair<string, string>[] userDefinedQueryParams;
+
+        /// <summary>Gets the user defined query parameters.</summary>
+        public KeyValuePair<string, string>[] UserDefinedQueryParams { get { return userDefinedQueryParams; } }

```

Comment:
Why not having IEnumerable<KeyValuePair<string, string>>? In general I prefer IEnumerable over primitive array.